torch/csrc/jit/backends/backend_init.cpp

```diff
@@ -234,6 +288,30 @@ void initJitBackendBindings(PyObject* module) {
                 py::cast<Module>(orig_module.attr("_c")),
                 method_compile_spec));
       });
+
+  m.def(
+      "_jit_to_backend_selective",
+      [=](const std::string& backend_name,
+          py::handle orig_module,
+          py::function to_backend,
+          const std::vector<std::string>& modules_to_lower) {
+        if (auto original_module =
+                as_module(py::cast<py::object>(orig_module))) {
+          // Clone the original module so that type modifications don't
+          // interfere with the changes that to_backend_selective_impl will
+          // make.
+          Module& mod = original_module.value();
+          auto cloned_mod = mod.clone();
```

Comment:
Hm, so this creates a module hierarchy with all submodule types cloned, but then you go in and swap the submodule types again to the lowered version. I think that will leave dangling references to the cloned submodule types in various method graphs. So if you had code that was like:
```
prim::CallMethod(%submod : ClonedType, "forward")
```
We haven't fixed up the cloned type to actually point to `LoweredType`. So the implementation of `CallMethod` will access the `ClonedType`'s forward in the interpreter instead of the new `LoweredType`'s version.

1. The type remapper in `clone_impl` is supposed to do the work of finding all references to type A and remap them to type B.
2. This suggests that we should write additional tests where the lowered and python versions of the module actually return *different* results, so that we can test that we are calling the right one.

Reply:
I added FileCheck tests for the post lowering graphs to make sure that types are replaced (although FileCheck is too coarse grained to do the kind of checks we need).